pom.xml

```
@@ -97,6 +97,8 @@
         <jacoco.version>0.8.7</jacoco.version>
         <jakarta-annotation-api.version>1.3.5</jakarta-annotation-api.version>
         <junit.version>4.13.2</junit.version>
```

Can we drop the JUnit 4 dependencies?

---

```suggestion
        <junit.version>4.13.2</junit.version>
```
junit dependency propagates hamcrest which is still used in odata.
[INFO] +- junit:junit:jar:4.13.2:test
[INFO] |  \- org.hamcrest:hamcrest-core:jar:2.2:test
[INFO] |     \- org.hamcrest:hamcrest:jar:2.2:test
Proposed solutions:
1) junit dependency can be replaced by hamcrest depenendency:
```
<dependency>
    <groupId>org.hamcrest</groupId>
    <artifactId>hamcrest</artifactId>
    <version>${hamcrest_version}</version>
    <scope>test</scope>
</dependency>
```
2) hamcrest assertions can be replaced with junit5 assertions, hamcrest and junit4 dependencies are not needed anymore

---

Yep - let's see if we can ditch hamcrest